### ISSUE TITLE
Could not load file or assembly Microsoft.Extensions.DependencyInjection.Abstractions

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule#2547]
**Description of the issue**

An exception is reported when before rules are run.

**Error output**

```text
OperationStopped: Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.
```

**Module in use and version:**

- Module: PSRule
- Version: **3.0.0-B0267**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.4.5
PSEdition                      Core
GitCommitId                    7.4.5
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Additional context**

This may be a conflict with Az.Resources v6.16.2 when running PSRule.Rules.Azure via PowerShell 7.0.

**Comment by BernieWhite** [comment]
Confirmed this is related to the current dependency chain of old versions on Az.Resources and will be addressed by https://github.com/Azure/PSRule.Rules.Azure/issues/2971.
